pkg/manifest/anaconda_installer_iso_tree.go

```diff
@@ -327,136 +333,114 @@ func (p *AnacondaInstallerISOTree) serialize() osbuild.Pipeline {
-		}
+	// Configure the kickstart file with the payload and any user options
+	kickstartOptions, err := osbuild.NewKickstartStageOptionsWithOSTreeCommit(
+		p.KSPath,
+		p.Users,
+		p.Groups,
+		makeISORootPath(p.PayloadPath),
+		p.ostreeCommitSpec.Ref,
+		p.Remote,
+		p.OSName)
+
+	if err != nil {
+		panic("failed to create kickstart stage options")
 	}
 
-	if p.containerSpec != nil {
-		images := osbuild.NewContainersInputForSources([]container.Spec{*p.containerSpec})
-
-		pipeline.AddStage(osbuild.NewMkdirStage(&osbuild.MkdirStageOptions{
-			Paths: []osbuild.MkdirStagePath{
-				{
-					Path: p.PayloadPath,
-				},
-			},
-		}))
+	stages = append(stages, p.makeKickstartStages(kickstartOptions)...)
 
-		// copy the container in
-		pipeline.AddStage(osbuild.NewSkopeoStageWithOCI(
-			p.PayloadPath,
-			images,
-			nil))
-
-		// do what we can in our kickstart stage
-		kickstartOptions, err := osbuild.NewKickstartStageOptionsWithOSTreeContainer(
-			"/osbuild-base.ks",
-			p.Users,
-			p.Groups,
-			path.Join("/run/install/repo", p.PayloadPath),
-			"oci",
-			"",
-			"")
+	return stages
+}
 
-		if err != nil {
-			panic("failed to create kickstartstage options")
-		}
+func (p *AnacondaInstallerISOTree) ostreeContainerStages() []*osbuild.Stage {
+	stages := make([]*osbuild.Stage, 0)
 
-		pipeline.AddStage(osbuild.NewKickstartStage(kickstartOptions))
+	images := osbuild.NewContainersInputForSources([]container.Spec{*p.containerSpec})
 
-		// and what we can't do in a separate kickstart that we include
-		targetContainerTransport := "registry"
-		if p.containerSpec.ContainersTransport != nil {
-			targetContainerTransport = *p.containerSpec.ContainersTransport
-		}
-		// Canonicalize to registry, as that's what the bootc stack wants
-		if targetContainerTransport == "docker://" {
-			targetContainerTransport = "registry"
-		}
+	stages = append(stages, osbuild.NewMkdirStage(&osbuild.MkdirStageOptions{
+		Paths: []osbuild.MkdirStagePath{
+			{
+				Path: p.PayloadPath,
+			},
+		},
+	}))
 
-		// Because osbuild core only supports a subset of options, we append to the
-		// base here with some more hardcoded defaults
-		// that should very likely become configurable.
-		hardcodedKickstartBits := `
-%include /run/install/repo/osbuild-base.ks
+	// copy the container in
+	stages = append(stages, osbuild.NewSkopeoStageWithOCI(
+		p.PayloadPath,
+		images,
+		nil))
+
+	// do what we can in our kickstart stage
+	kickstartOptions, err := osbuild.NewKickstartStageOptionsWithOSTreeContainer(
+		p.KSPath,
+		p.Users,
+		p.Groups,
+		path.Join("/run/install/repo", p.PayloadPath),
+		"oci",
+		"",
+		"")
+
+	// NOTE: these are similar to the unattended kickstart options in the
+	// other two payload configurations but partitioning is different and
+	// we need to add that separately, so we can't use makeKickstartStage
+	kickstartOptions.RootPassword = &osbuild.RootPasswordOptions{
+		Lock: true,
+	}
+	kickstartOptions.Lang = "en_US.UTF-8"
```

Comment:
No they're just a set of defaults we need to decide on.  Right now they're the ones that came from the previous PRs for the RHEL for Edge installer.  I would much prefer that each distro / image type define their own defaults and we would fail if unattended is supported without them being defined, but for now we need some "lowest level" fallbacks.